clients/mediaconvert.go

```diff
@@ -0,0 +1,155 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/mediaconvert"
+)
+
+// TODO: Real configs here. Only a placeholder for now from what I think is the
+// required stuff. There maybe more or just be different.
+type MediaConvertOptions struct {
+	/*add any other config you need like endpoint, role, queue etc*/
+	Endpoint, Region, Role       string
+	AccessKeyID, AccessKeySecret string
+}
+
+type MediaConvert struct {
+	// TODO
+}
+
+func NewMediaConvertClient(opts MediaConvertOptions) TranscodeProvider {
+	return &MediaConvert{
+		// TODO
+	}
+}
+
+// This should do the whole transcode job, including the polling loop for the
+// job status until it is completed.
+//
+// It should call the input.ReportProgress function to report the progress of
+// the job during the polling loop.
+func (mc *MediaConvert) Transcode(ctx context.Context, input TranscodeJobInput) error {
+	config := aws.Config{
+		Region:      aws.String("us-east-1"),
+		Credentials: credentials.NewSharedCredentials("", "default"),
+	}
+	sess, err := session.NewSession(&config)
+
+	svc := mediaconvert.New(sess)
+
+	// Add endpoint to the service client
+	svc.Endpoint = "https://vasjpylpa.mediaconvert.us-east-1.amazonaws.com"
```

Comment:
The strings here should all come from the client config instead of being hardcoded. Check the new options struct I created. Also keep in mind that you should use static credentials [like this](https://github.com/livepeer/go-tools/blob/db20184c5f12055422db3c1b1447b6f05f64c99e/drivers/s3.go#L117) instead of shared ones which are not available in production. 

The session and service objects should also not be created on every run. They should be stored in the client struct instead and reused for every request.